operator/pkg/util/yaml.go

```diff
@@ -23,13 +23,52 @@ import (
 	"strings"
 
 	jsonpatch "github.com/evanphx/json-patch/v5"
-	"github.com/gogo/protobuf/jsonpb"
-	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/jsonpb"
+	legacyproto "github.com/golang/protobuf/proto"
 	"github.com/kylelemons/godebug/diff"
+	"google.golang.org/protobuf/proto"
 	yaml3 "k8s.io/apimachinery/pkg/util/yaml"
 	"sigs.k8s.io/yaml"
+
+	"istio.io/istio/pkg/util/protomarshal"
 )
 
+func ToYAMLGeneric(root interface{}) ([]byte, error) {
+	var vs []byte
+	if proto, ok := root.(proto.Message); ok {
+		v, err := protomarshal.ToYAML(proto)
+		if err != nil {
+			return nil, err
+		}
+		vs = []byte(v)
+	} else {
+		v, err := yaml.Marshal(root)
+		if err != nil {
+			return nil, err
+		}
+		vs = v
+	}
+	return vs, nil
+}
+
+func MustToYAMLGeneric(root interface{}) string {
```

Comment:
why not just call the other method and panic?

Reply:
I don't really have a strong opinion - I am not sure if this can be triggered from user input where panic is unsafe?

This is preserving existing functionality - my goal in this PR is to have 0 user facing behavior